File: Axity/benchmarks/python/loops.py
#!/usr/bin/env python3

# Loop operations benchmark in Python

def sum_with_for_loop(n):
    total = 0
    for i in range(n):
        total += i
    return total

def sum_with_while_loop(n):
    total = 0
    i = 0
    while i < n:
        total += i
        i += 1
    return total

def nested_loops(depth, size):
    count = 0
    for i in range(depth):
        for j in range(size):
            for k in range(size):
                count += 1
    return count

# Perform loop operations
for_result = sum_with_for_loop(100000)
while_result = sum_with_while_loop(100000)
nested_result = nested_loops(10, 50)

print(f"Loops: for={for_result} while={while_result} nested={nested_result}")

File: benchmarks/javascript/loops.js
#!/usr/bin/env bun

// Loop operations benchmark in JavaScript

function sumWithForLoop(n) {
    let total = 0;
    for (let i = 0; i < n; i++) {
        total += i;
    }
    return total;
}

function sumWithWhileLoop(n) {
    let total = 0;
    let i = 0;
    while (i < n) {
        total += i;
        i++;
    }
    return total;
}

function nestedLoops(depth, size) {
    let count = 0;
    for (let i = 0; i < depth; i++) {
        for (let j = 0; j < size; j++) {
            for (let k = 0; k < size; k++) {
                count++;
            }
        }
    }
    return count;
}

// Perform loop operations
const forResult = sumWithForLoop(100000);
const whileResult = sumWithWhileLoop(100000);
const nestedResult = nestedLoops(10, 50);

console.log(`Loops: for=${forResult} while=${whileResult} nested=${nestedResult}`);